File: app/src/main/java/fr/suylo/myappkotlin/MainActivity.java
package fr.suylo.myappkotlin;

import android.net.Uri;
import android.util.Log;
import android.widget.ImageView;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.squareup.picasso.Picasso;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import java.net.MalformedURLException;
import java.net.URI;
import java.net.URISyntaxException;
import java.net.URL;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private TextView zeTextView;
    private PLaceHolderAPI zePlaceHolderAPI;
    public String content;
    ImageView zeImageView = findViewById(R.id.imageView2);


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_apiactivity);
        zeTextView = findViewById(R.id.textdefou);
        Retrofit retrofit = new Retrofit.Builder().
                baseUrl("https://api.chucknorris.io/jokes/").
                addConverterFactory(GsonConverterFactory.create()).
                build();
        zePlaceHolderAPI = retrofit.create(PLaceHolderAPI.class);
        Call<PlaceHolderPost> call = zePlaceHolderAPI.getPosts();
        call.enqueue(

                new Callback<PlaceHolderPost>() {
                    @Override
                    public void onResponse(Call<PlaceHolderPost> call, Response<PlaceHolderPost> response) {
                        if (response.isSuccessful()) {
                            PlaceHolderPost p = response.body();
                            content = p.getValue();
                        }
                    }
                    @Override
                    public void onFailure(Call<PlaceHolderPost> call, Throwable t) {
                        zeTextView.setText(t.getMessage());
                    }
                }
        );
        content.replaceAll(" ", "_");
        Picasso.get().load("https://api.memegen.link/images/aag/"+ content + "/e.png").into(zeImageView);
    }
}

File: app/src/main/java/fr/suylo/myappkotlin/MainActivity.kt
package fr.suylo.myappkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import java.util.ArrayList

class MainActivity : AppCompatActivity() {
    val arrayList = ArrayList<Int>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d("MyApp", "Test des logs")

        this.arrayList.add(1)
        this.arrayList.add(2)
        this.arrayList.add(3)
    }

    fun onClickText(view: View) {
        val answers = arrayOf<String>()
        val question = findViewById<TextView>(R.id.questionText)
        val askOne = findViewById<TextView>(R.id.answer_1)
        val askTwo = findViewById<TextView>(R.id.answer_2)
        val askThree = findViewById<TextView>(R.id.answer_3)
        val pgBar = findViewById<ProgressBar>(R.id.progressBar)


        for (i in 1..this.arrayList.size) {
            // TODO: faut réfléchir un ptit peu là
        }

        Log.d("MyApp", "OnClick Element text");
    }
}